src/lexer_tok.c

```diff
@@ -0,0 +1,136 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   lexer_tok.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/02 18:51:59 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/12 22:46:04 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+#include "lexer_tok.h"
+
+/**
+ * @brief Create a token.
+ * 
+ * Token contains the token string and its length.
+ * @param token 	Created token.
+ * @param s 		Token string.
+ * @return t_error 	ERR_MALLOC, SUCCESS
+ */
+t_error	ft_create_tok(t_tok *token, char *s)
+{
+	if (!token || !s)
+		return (ERR_EMPTY);
+	token->tok_size = ft_strlen(s);
+	token->tok = ft_realloc (token->tok, token->tok_size + 1, 1);
+	if (!token->tok)
+	{
+		free(token->tok);
+		return (ERR_MALLOC);
+	}
+	ft_memcpy(token->tok, s, token->tok_size);
+	token->tok[token->tok_size] = '\0';
+	return (SUCCESS);
+}
+
+/**
+ * @brief Overarching function to create a token.
+ * 
+ * Divides input string into separate tokens. The token string
+ * is extracted into a buffer string before being saved inside
+ * a token.
+ * @param src 		Struct containing the input string,
+ * 					its length and current position.
+ * @param token 	Token to be created.
+ * @return t_error* ERR_EMPTY, ERR_MALLOC, SUCCESS
+ */
+t_error	ft_tokenise(t_src *src, t_tok *token)
+{
+	char	*buf;
+	int		buf_size;
+	int		buf_pos;
+	t_error	err;
+
+	if (!src || !src->buf || !src->buf_size)
+		return (ERR_EMPTY);
+	buf_size = 2048;
```

Comment:
We could have the buffer outside of the tokenise function. Right now for every created token the buffer has to be malloc'd. If we could we reuse him, that would be fancy

Reply:
No immediate benefit, therefore in backlog